### PR TITLE
Fix_sg_attachment

### PR DESCRIPTION
**Pull request by herman-wong-cf**
The presence of the aws_network_interface_sg_attachment resource conflicts with the "vpc_security_group_ids" parameter in the aws_instance resource.  When you attempt to include "additional_security_group_ids", it causes a perpetual diff that removes the additional security group attachment when reapplying this module.  Simply removing the vpc_security_group_ids parameter fixes the issue.

### CODE REVIEW

**Comment by herman-wong-cf** [comment]
Initial testing by adding "base_win_sg_id" to a Windows bastion:
image

Applying this module a second time causes the additional security group id (base_win_sg_id) to be removed, because it isn't part of the "vpc_security_group_ids" parameter:
image

Removing this parameter fixes the behavior:
image

This fix was tested on both cases where an additional security group id was provided and one where it isn't used, both cases will not result in perpetual diffs or any other problems.

This fix is required to use "Base Security Groups" where you might want to add common ports (22 for SSH, 5986 for WinRM), or add global ingress for scanning (139 and 445 for Nessus).

**Comment by herman-wong-cf** [comment]
May need to hold off and rework, new EC2 module deployment does not result in ANY SG attachment other than a default (no rules).

**Comment by herman-wong-cf** [comment]
Added conditional to fix SG attachment, works without issue without additional SG, will need to confirm with additional SG

**Comment by herman-wong-cf** [comment]
Prior code did not work with additional SG:
image

**Comment by herman-wong-cf** [comment]
The SG attachment to primary ENI resource seems to be the primary cause of the issues, removing that and using a conditional + concat() with the var.additional_security_groups appears to resolve the issue.

Tested new code under 2 scenarios:
- Nexpose without additional SG, no change upon reapply.
- Windows bastion with additional SG (base-win-mgmt-sg), no change upon reapply, base SG rules are attached.

**Comment by herman-wong-cf** [comment]
Deployed Palo module as that makes use of additional ENIs to ensure that aspect wasn't negatively impacted:
image

**Comment by az-kennedy** [comment]
@herman-wong-cf , let's update the CODEOWNERS file as part of this PR.

az-kennedy and your github account. THX

**Review by az-kennedy — APPROVED** [review]
lgtm

**Pull request merged**
